cfme/tests/infrastructure/test_individual_host_creds.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+import pytest
+import random
+import utils.conf as conf
+import cfme.web_ui.flash as flash
+
+from cfme.infrastructure import host
+from cfme.infrastructure.provider import details_page
+from cfme.web_ui import fill
+from utils import testgen
+from utils.blockers import BZ
+from utils.update import update
+
+msgs = {
+    'virtualcenter': 'Cannot complete login due to an incorrect user name or password.',
+    'rhevm': 'Login failed due to a bad username or password.'
+}
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(metafunc, ['virtualcenter', 'rhevm'])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+def get_host_data_by_name(provider_key, host_name):
+    for host_obj in conf.cfme_data.get('management_systems', {})[provider_key].get('hosts', []):
+        if host_name == host_obj['name']:
+            return host_obj
+    return None
+
+
+def test_host_good_creds(request, setup_provider, provider):
+    """
+    Tests host credentialing  with good credentials
+    """
+    test_host = random.choice(provider.get_yaml_data()["hosts"])
+    host_data = get_host_data_by_name(provider.key, test_host.name)
+    host_obj = host.Host(name=test_host.name)
+
+    host_obj.update(
+        updates={'credentials': host.get_credentials_from_config(host_data['credentials'])},
+        validate_credentials=True
+    )
+
+    flash.assert_message_match('Host "{}" was saved'.format(test_host.name))
+
+    # Remove creds after test
+    @request.addfinalizer
+    def _host_remove_creds():
+        with update(host_obj):
+            host_obj.credentials = host.Host.Credential(
+                principal="", secret="", verify_secret="")
+
+
+@pytest.mark.meta(
+    blockers=BZ(1310910, unblock=lambda provider: provider.type != 'rhevm'),
+)
+def test_host_bad_creds(request, setup_provider, provider):
+    """
+    Tests host credentialing  with bad credentials
+    """
+    pytest.sel.force_navigate('infrastructure_provider', context={'provider': provider})
+    pytest.sel.click(details_page.infoblock.element("Relationships", "Hosts"))
+
+    host_name = random.choice(provider.get_yaml_data()["hosts"])
+    pytest.sel.force_navigate('infrastructure_host_edit', context={
+        'host': host_name, })
+    creds = conf.credentials['bad_credentials']
+    fill(host.credential_form, {'default_principal': creds['username'],
+                           'default_secret': creds['password'],
```

Comment:
I've created a host object for the test_host_good_creds test. 

I haven't instantiated a host object for the test_host_bad_creds test.In the fill method in infrastructure/host.py,we are already asserting that there are no flash error messages,because of which test_host_bad_creds would always fail.